src/main/java/com/openlattice/chronicle/controllers/v2/ChronicleControllerV2.java

```diff
@@ -180,6 +183,22 @@ public void sendMessages(
         messageService.sendMessages( organizationId, messageDetails);
     }
 
+    @RequestMapping(
+            path = { ORGANIZATION_ID_PATH + MESSAGE_PATH + STATUS_PATH },
+            method = RequestMethod.POST )
+    @ResponseStatus( HttpStatus.OK )
+    public void updateMessageStatus(
+            @PathVariable ( ORGANIZATION_ID ) UUID organizationId,
+            HttpServletRequest request
+    ) throws ExecutionException {
+        String messageSid = request.getParameter( "MessageId ");
+        String status = request.getParameter( "MessageStatus" );
+
+        if ( status.equals( "undelivered" ) || status.equals( "failed" ) ) {
+            messageService.trackUndeliveredMessage( organizationId, messageSid );
+        }
```

Comment:
`undelivered` and `failed` should be enums, perhaps `MessageStatus`? Also, if you create an enum, then the above suggestion would be ...

```
@RequestParam( MESSAGE_STATUS ) MessageStatus messageStatus	
```